storage/client.go

```diff
@@ -356,6 +400,66 @@ func (c Client) exec(verb, url string, headers map[string]string, body io.Reader
 		body:       resp.Body}, nil
 }
 
+func (c Client) execInternalJSON(verb, url string, headers map[string]string, body io.Reader) (*odataResponse, error) {
+	req, err := http.NewRequest(verb, url, body)
+	for k, v := range headers {
+		req.Header.Add(k, v)
+	}
+
+	httpClient := http.Client{}
```

Comment:
hey @MindFlavor I just merged #298 which reads this from `Client.HTTPClient` (and defaults to `net/http#DefaultClient`). In this case you'll probably need to rebase your code on top of `master` and refactor this part sometime.